MetaMask/metamask-mobile
# fix: PPOM - Metrics information from ppom is not logged

## **Description**

Whenever we trigger a transaction from a dapp, we don't see any information about Blockaid validation. This is because the blockaid banner now shows loading initially for some time before the actual alert is shown. We were sending metrics only on component load, but we should send metrics for blockaid alert

The solution used here is to delay metrics for blockaid until user clicks confirm or cancel. At that point we have a definitive blockaid status to send (if it's loading, we send loading, if it's loaded we send the normal metrics).

For the deleted test cases in MessageSign (because enzyme does not support useEffect). I had a call with @cryptotavares and we will convert the tests to RTL in another PR and add the required useEffect tests there. There's no need to delay this because of that.

## **Related issues**

Fixes: #7795 

## **Manual testing steps**

1. Checkout this branch and run MetaMask Mobile
2. Enable blockaid in settings
4. Open testdapp and perform a PPOM transaction
5. Check the console and notice how security alert metrics is not sent
6. Now click/confirm cancel while the banner is still showing loading and  check console to see how security alert metrics is sent with loading
7. Now perform another PPOM transaction, this time wait for the blockaid banner to fully load and then click confirm/cancel and see how the security alert response is sent in metrics.

## **Screenshots/Recordings**



### **Before**
[](https://github.com/MetaMask/metamask-mobile/assets/54408225/a918aa35-a881-46bb-a839-ac62e01e814d)

### **After**
https://github.com/MetaMask/metamask-mobile/assets/44811/3fb7f759-a1c7-4eaa-b42a-51243d6b340b

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
